### PR TITLE
Move the StorageVersions types to kube-apiserver

### PR DESCRIPTION
**Pull request by roycaihw**
/kind api-change

Move the internal types to kube-apiserver, and the external types to k8s.io/api. Ref https://github.com/kubernetes/kubernetes/pull/92064#discussion_r447795942

The first three commits are based on https://github.com/kubernetes/kubernetes/pull/93164

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig api-machinery
/priority important-soon
/assign @caesarxuchao

### CODE REVIEW

File: staging/src/k8s.io/api/apiserverinternal/v1alpha1/doc.go [hunk 1]
```diff
@@ -16,12 +16,8 @@ limitations under the License.
 
 // +k8s:deepcopy-gen=package
 // +k8s:protobuf-gen=package
-// +k8s:conversion-gen=k8s.io/apiserver/pkg/apis/apiserverinternal
 // +k8s:openapi-gen=true
-// +k8s:defaulter-gen=TypeMeta
-
-// +groupName=internal.apiserver.k8s.io
```
Comment (caesarxuchao) [inline staging/src/k8s.io/api/apiserverinternal/v1alpha1/doc.go:23]: I expect this line is kept, like https://github.com/kubernetes/kubernetes/blob/adefb2ebc8866bac3b17a6e7bbdf33a31128ec3d/staging/src/k8s.io/api/flowcontrol/v1alpha1/doc.go#L21

File: staging/src/k8s.io/client-go/informers/generic.go [hunk 2]
```diff
@@ -100,6 +101,10 @@ func (f *sharedInformerFactory) ForResource(resource schema.GroupVersionResource
 	case v1beta1.SchemeGroupVersion.WithResource("validatingwebhookconfigurations"):
 		return &genericInformer{resource: resource.GroupResource(), informer: f.Admissionregistration().V1beta1().ValidatingWebhookConfigurations().Informer()}, nil
 
+		// Group=apiserverinternal, Version=v1alpha1
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/informers/generic.go:104]: `Group=apiserverinternal` is wrong.
I think if you keep the groupName tag, the comment here would be correct.

File: staging/src/k8s.io/client-go/kubernetes/typed/apiserverinternal/v1alpha1/fake/fake_storageversion.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	watch "k8s.io/apimachinery/pkg/watch"
+	testing "k8s.io/client-go/testing"
+)
+
+// FakeStorageVersions implements StorageVersionInterface
+type FakeStorageVersions struct {
+	Fake *FakeApiserverinternalV1alpha1
+}
+
+var storageversionsResource = schema.GroupVersionResource{Group: "apiserverinternal", Version: "v1alpha1", Resource: "storageversions"}
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/kubernetes/typed/apiserverinternal/v1alpha1/fake/fake_storageversion.go:38]: The group here is wrong.

File: staging/src/k8s.io/client-go/kubernetes/typed/apiserverinternal/v1alpha1/fake/fake_storageversion.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+var storageversionsKind = schema.GroupVersionKind{Group: "apiserverinternal", Version: "v1alpha1", Kind: "StorageVersion"}
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/kubernetes/typed/apiserverinternal/v1alpha1/fake/fake_storageversion.go:40]: The Group here is wrong.

**Review by caesarxuchao — COMMENTED** [review]
Otherwise lgtm.

**Comment by caesarxuchao** [comment]
A nit, otherwise lgtm.

**Comment by roycaihw** [comment]
Gentle ping

**Comment by liggitt** [comment]
add to:
```
diff --git a/staging/src/k8s.io/api/roundtrip_test.go b/staging/src/k8s.io/api/roundtrip_test.go
index beafa9d1a25..0bc318a5da4 100644
--- a/staging/src/k8s.io/api/roundtrip_test.go
+++ b/staging/src/k8s.io/api/roundtrip_test.go
@@ -24,6 +24,7 @@ import (
 	admissionv1beta1 "k8s.io/api/admission/v1beta1"
 	admissionregv1 "k8s.io/api/admissionregistration/v1"
 	admissionregv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	apiserverinternalv1alpha1 "k8s.io/api/apiserverinternal/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
 	appsv1beta1 "k8s.io/api/apps/v1beta1"
 	appsv1beta2 "k8s.io/api/apps/v1beta2"
@@ -75,6 +76,7 @@ var groups = []runtime.SchemeBuilder{
 	admissionv1.SchemeBuilder,
 	admissionregv1beta1.SchemeBuilder,
 	admissionregv1.SchemeBuilder,
+	apiserverinternalv1alpha1.SchemeBuilder,
 	appsv1beta1.SchemeBuilder,
 	appsv1beta2.SchemeBuilder,
 	appsv1.SchemeBuilder,
```

and generate fixtures with `UPDATE_COMPATIBILITY_FIXTURE_DATA=true go test k8s.io/api` for the generated commit

lgtm otherwise

**Comment by liggitt** [comment]
/approve
for API file move

/hold
for [test data comment](https://github.com/kubernetes/kubernetes/pull/93087#issuecomment-666046074)

**Comment by liggitt** [comment]
looking at the follow-up PR, I'm also confused as to whether kube-apiserver or kube-aggregator is the hosting server for this API, xref https://github.com/kubernetes/kubernetes/pull/92064#issuecomment-666050323

**Comment by roycaihw** [comment]
updated testdata and squashed. Please take a look
```
commit 20f6784dd129438ca28b45c8f16939285ead56bc (HEAD -> storage-version/move-types)
Author: Haowei Cai <haoweic@google.com>
Date:   Thu Jul 30 10:06:29 2020 -0700

    fixup! generated

 staging/src/k8s.io/api/testdata/HEAD/internal.apiserver.k8s.io.v1alpha1.StorageVersion.json |  66 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 staging/src/k8s.io/api/testdata/HEAD/internal.apiserver.k8s.io.v1alpha1.StorageVersion.pb   | Bin 0 -> 350 bytes
 staging/src/k8s.io/api/testdata/HEAD/internal.apiserver.k8s.io.v1alpha1.StorageVersion.yaml |  46 ++++++++++++++++++++++++++++++++++++++++++++++
 3 files changed, 112 insertions(+)

commit cc587547599d67486d7e675d0b7acee1aee7b99f
Author: Haowei Cai <haoweic@google.com>
Date:   Thu Jul 30 10:06:10 2020 -0700

    fixup! fix import paths

 staging/src/k8s.io/api/roundtrip_test.go | 2 ++
 1 file changed, 2 insertions(+)
```

**Comment by roycaihw** [comment]
> whether kube-apiserver or kube-aggregator is the hosting server for this API

The commit `Let kube-aggregator host the storage version API ` was from #84016. I cherrypicked @caesarxuchao 's commit and started from there. The host server is kube-apiserver

**Comment by roycaihw** [comment]
resolved the conflict in `hack/.golint_failures`

**Pull request merged**